src/urdf_parser_py/urdf.py

```diff
@@ -533,6 +533,20 @@ def get_root(self):
         assert root is not None, "No roots detected, invalid URDF."
         return root
 
+    def post_read_xml(self):
```

Comment:
What calls this function? I can't find any uses of it.

Reply:
It's magic :).  The `Robot` class is a sub-class of `xmlr.Object`, which defines an empty version of `post_read_xml`.  In turn, this method is called directly after reading in the XML (see [here](https://github.com/ros/urdf_parser_py/blob/b213e6e5846800b4873be349d605b906e78d681e/src/urdf_parser_py/xml_reflection/core.py#L590)).  So I just overrode it in the sub-class.